microsoft/vscode
# Add an option to expose the allowMidWordMatch as a setting

This PR fixes #53715 

This won't change the default experience but will give people an option to change the fuzzy search algorithm that is used. 

How to test:
1. Create a new C# test.cs file
2. install omnisharp
3. Populate the file with this

```cs
using System;
using System.Collections.Generic;

public class FuzzyTest
{

    public FuzzyTest()
    {
        // Type here
    }
}

public class FBUserSession
{
    string test;
    string test2;
}
```
4. In constructor, type FBUserSession, UserSession and Session. The first and last will work with the setting off.
5. Turn the new setting on 
6. Redo test for step 4 and see that all those words typed will return the FBUserSession object that the user wants
